src/plugins/orangehrmClaimPlugin/Api/ClaimEventAPI.php

```diff
@@ -0,0 +1,257 @@
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Claim\Api;
+
+use Exception;
+use OrangeHRM\Claim\Api\Model\ClaimEventModel;
+use OrangeHRM\Claim\Dto\ClaimEventSearchFilterParams;
+use OrangeHRM\Claim\Traits\Service\ClaimServiceTrait;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Exception\BadRequestException;
+use OrangeHRM\Core\Api\V2\Exception\InvalidParamException;
+use OrangeHRM\Core\Api\V2\Exception\RecordNotFoundException;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Serializer\NormalizeException;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use OrangeHRM\Entity\ClaimEvent;
+use OrangeHRM\ORM\Exception\TransactionException;
+
+class ClaimEventAPI extends Endpoint implements CrudEndpoint
+{
+    use EntityManagerHelperTrait;
+    use ClaimServiceTrait;
+
+    public const PARAMETER_NAME = 'name';
+    public const PARAMETER_DESCRIPTION = 'description';
+    public const PARAMETER_ID = 'id';
+    public const PARAMETER_IDS = 'ids';
+    public const PARAMETER_STATUS = 'status';
+    public const DESCRIPTION_MAX_LENGTH = 1000;
+    public const NAME_MAX_LENGTH = 100;
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        try {
+            $claimEvent = new ClaimEvent();
+            $this->setClaimEvent($claimEvent);
+            return new EndpointResourceResult(ClaimEventModel::class, $claimEvent);
+        } catch (InvalidParamException|BadRequestException $e) {
+            throw $e;
+        } catch (Exception $e) {
+            throw new TransactionException($e);
+        }
+    }
+
+    /**
+     * @param ClaimEvent $claimEvent
+     */
+    public function setClaimEvent(ClaimEvent $claimEvent)
+    {
+        $name = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        $description = $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_DESCRIPTION);
+        $status = $this->getRequestParams()->getBoolean(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_STATUS, true);
+        $claimEvent->setName($name);
+        $claimEvent->setDescription($description);
+        $claimEvent->setStatus($status);
+        $this->getClaimEventService()->getClaimEventDao()->saveEvent($claimEvent);
```

Comment:
Same event cannot add multiple times. 
_"Already Exists" validation should show up when trying to save an event that is already present in the list_